File: LocationSystem.java
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class LocationSystem{

    private List<Item> items;

    public LocationSystem(List<Item> itemNames) {
        this.items = new ArrayList<>();
        for (Item name : itemNames) {
            items.add(name);
        }
    }

    public void showAvailableItems() {
        System.out.println("Articles disponibles :");
        items.stream()
            .filter(item -> item.getStatus().equals(StatusEnum.DISPONIBLE))
            .forEach(item -> System.out.println("- " + item.getName()));
    }

    public void showRentedItems() {
        System.out.println("Articles loués :");
        items.stream()
            .filter(item -> item.getStatus().equals(StatusEnum.LOUÉ))
            .forEach(item -> System.out.println("- " + item.getName()));
    }

    public void rentItem(String itemName, int durationInDays) {
        Item item = items.stream()
            .filter(i -> i.getName().equalsIgnoreCase(itemName) && i.getStatus().equals(StatusEnum.DISPONIBLE))
            .findFirst()
            .orElse(null);

        if (item == null) {
            System.out.println("L'article \"" + itemName + "\" n'est pas disponible pour le moment.");
            return;
        }

        String returnDate = calculateReturnDate(durationInDays);
        item.rent(returnDate);
        System.out.println("L'article \"" + itemName + "\" a été loué pour " + durationInDays + " jour(s). Retour prévu le " + returnDate + ".");
    }

    public void returnItem(String itemName) {
        Item item = items.stream()
            .filter(i -> i.getName().equalsIgnoreCase(itemName) && i.getStatus().equals(StatusEnum.LOUÉ))
            .findFirst()
            .orElse(null);

        if (item == null) {
            System.out.println("L'article \"" + itemName + "\" n'a pas été trouvé parmi les articles loués.");
            return;
        }

        item.returnItem();
        System.out.println("L'article \"" + itemName + "\" a été retourné et est maintenant disponible.");
    }

    private String calculateReturnDate(int durationInDays) {
        SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yyyy");
        Date today = new Date();
        Date returnDate = new Date(today.getTime() + (long) durationInDays * 24 * 60 * 60 * 1000);
        return sdf.format(returnDate);
    }

}

File: LocationSystem.kt
import java.text.SimpleDateFormat
import java.util.Date

class LocationSystem(private val items: MutableList<Item>) {

    fun showAvailableItems() {
        println("Articles disponibles :")
        items.filter { it.getStatus() == Status.DISPONIBLE }
            .forEach { println("- ${it.getName()}") }
    }

    fun showRentedItems() {
        println("Articles loués :")
        items.filter { it.getStatus() == Status.LOUE }
            .forEach { println("- $it") }
    }

    fun rentItem(itemName: String, durationInDays: Int) {
        val item = items.find { it.getName().equals(itemName, ignoreCase = true) && it.getStatus() == Status.DISPONIBLE }

        if (item == null) {
            println("L'article \"$itemName\" n'est pas disponible pour le moment.")
            return
        }

        val returnDate = calculateReturnDate(durationInDays)
        item.rent(returnDate)
        println("L'article \"$itemName\" a été loué pour $durationInDays jour(s). Retour prévu le $returnDate.")
    }

    fun returnItem(itemName: String) {
        val item = items.find { it.getName().equals(itemName, ignoreCase = true) && it.getStatus() == Status.LOUE }

        if (item == null) {
            println("L'article \"$itemName\" n'a pas été trouvé parmi les articles loués.")
            return
        }

        item.returnItem()
        println("L'article \"$itemName\" a été retourné et est maintenant disponible.")
    }

    private fun calculateReturnDate(durationInDays: Int): String {
        val dateFormat = SimpleDateFormat("dd/MM/yyyy")
        val today = Date()
        val returnDate = Date(today.time + durationInDays * 24 * 60 * 60 * 1000L)
        return dateFormat.format(returnDate)
    }
}
